packages/runtime/test/integration/pool-transaction-fee-test.ts

```diff
@@ -86,4 +97,291 @@ describe("Pooled Transaction Fees Test", function () {
+				sign: types.SignType.SecretKey,
+				fromAccount: john.account,
+				toAccountAddr: alice.address,
+				amountMicroAlgos: amount,
+				payFlags: { totalFee: 0 }, // with 0 txn fee
+			},
+			{
+				type: types.TransactionType.TransferAlgo,
+				sign: types.SignType.SecretKey,
+				fromAccount: alice.account,
+				toAccountAddr: elon.address,
+				amountMicroAlgos: amount,
+				payFlags: { totalFee: fee }, // this covers fee of entire group txns
+			}
+		];
+
+		runtime.executeTx(groupTx)
+	});
+
+});
+
+describe("Pooled Transaction Fees Test with App and Asset", function () {
+	useFixture("stateful");
+	const initialBalance = 1e30;
+	let john: AccountStore;
+	let bob: AccountStore;
+	let elon: AccountStore;
+	let alice: AccountStore;
+	let assetId: number;
+	let approvalProgramFilename: string;
+	let clearProgramFilename: string;
+	let runtime: Runtime;
+
+	this.beforeEach(function () {
+		john = new AccountStore(initialBalance, "john");
+		bob = new AccountStore(initialBalance, "bob");
+		alice = new AccountStore(initialBalance, "alice");
+		elon = new AccountStore(0, "elon");
+		runtime = new Runtime([john, bob, alice, elon]); // setup test
+
+		approvalProgramFilename = "counter-approval.teal";
+		clearProgramFilename = "clear.teal";
+	});
+
+	function setupAsset(): void {
+		// create asset
+		assetId = runtime.deployASA("gold", {
+			creator: { ...john.account, name: "john" },
+		}).assetIndex;
+	}
+
+	function setupApp(): void {
+		// deploy new app
+		runtime.deployApp(
+			john.account,
+			{
+				appName: "app",
+				metaType: types.MetaType.FILE,
+				approvalProgramFilename,
+				clearProgramFilename,
+				globalBytes: 1,
+				globalInts: 1,
+				localBytes: 1,
+				localInts: 1,
+			},
+			{}
+		);
+	}
+
+	// helper function
+	function syncAccounts(): void {
+		john = runtime.getAccount(john.address);
+		bob = runtime.getAccount(bob.address);
+		alice = runtime.getAccount(alice.address);
+		elon = runtime.getAccount(elon.address);
+	}
+
+	it("Should not fail when tried to optin to unfunded account in group txn", function () {
+		setupAsset();
+		const amount = 200000;
+		const fee = 3000;
+		// group with fee distribution
+		const groupTx: types.ExecParams[] = [
+			{
+				type: types.TransactionType.TransferAlgo,
+				sign: types.SignType.SecretKey,
+				fromAccount: john.account,
+				toAccountAddr: alice.address,
+				amountMicroAlgos: amount,
+				payFlags: { totalFee: 0 }, // with 0 txn fee
+			},
+			{
+				type: types.TransactionType.TransferAlgo,
+				sign: types.SignType.SecretKey,
+				fromAccount: alice.account,
+				toAccountAddr: bob.address,
+				amountMicroAlgos: amount,
+				payFlags: { totalFee: fee }, // this covers fee of entire group txns
+			},
+			{
+				type: types.TransactionType.OptInASA,
+				sign: types.SignType.SecretKey,
+				fromAccount: elon.account, // unfunded account and no fund is sent to this account
+				assetID: assetId,
+				payFlags: { totalFee: 0 }, // with 0 txn fee
+			},
+		];
```

Comment:
again lets add asserts before and after. We can even add assert that it doesn't throw an error

Reply:
Done